src/app/view/timestamp-editor/timestamp-editor.component.ts

```diff
@@ -324,4 +329,14 @@ export class TimestampEditorComponent implements OnInit {
     return (String('0').repeat(2) + item).substr((2 * -1), 2);
   }
 
+  isDimensionUnit() {
+    if (this?.fullVesselDetails?.dimensionUnit) {
+      this.timestampFormGroup.controls.vesselDraft.enable();
+    } else {
+      this.timestampFormGroup.controls.vesselDraft.disable();
+    }
+    this.timestampFormGroup.controls.vesselDraft.updateValueAndValidity();
+    return this?.fullVesselDetails?.dimensionUnit ??  null; 
+  }
```

Comment:
It is named like a getter but it has a side-effect.  That is confusing and the side-effect logic should be moved elsewhere.

With the side-effect moved out (e.g., into ngInit), I think it should be renamed to `hasDimensionUnit` because that is what it really denotes (It is not clear what `isDimensionUnit` means - whereas `hasDimensionUnit` conveys a meaning).

Reply:
I agree, will rename now. Thanks :)

Reply:
The side-effects have not been fixed.